src/plugin.js

```diff
@@ -7,17 +7,13 @@
  * of a Microcosm.
  */
 
-function PluginFactory (config, options, app) {
-  return Object.assign({ app, options }, config)
-}
-
 function checkPlugin (plugin) {
-  if (process.env.NODE_ENV === 'development' && 'register' in plugin && typeof plugin.register !== 'function') {
-    throw TypeError('Expected register property of plugin to be a function, instead got ' + plugin.register)
+  if (process.env.NODE_ENV !== 'production' && typeof plugin !== 'function' && ('register' in plugin && typeof plugin.register !== 'function')) {
```

Comment:
Unrelated to this PR, and I'm working on a thing now related to this, but I don't like these types of `env` checks in all the places. The app should have a `debug` mode that could optional be configured to equal `process.env.NODE_ENV !== 'production'`, but alternatively could be anything our hearts desire.

Just a random thought.